charts/opensearch/templates/statefulset.yaml

```diff
@@ -57,6 +57,9 @@ spec:
         {{- if .Values.config }}
         configchecksum: {{ include (print .Template.BasePath "/configmap.yaml") . | sha256sum | trunc 63 }}
         {{- end }}
+        {{- if .Values.securityConfig.config.data }}
+        securityconfigchecksum: {{ include (print .Template.BasePath "/securityconfig.yaml") . | sha256sum | trunc 63 }}
+        {{- end }}
```

Comment:
There are other conditions where this basePath may be used. It'd be nice to have an `or` here to add in when the existing secret mount has been presented.

Reply:
It is only in the case when securityConfig.config.data is defined. This condition is from here:

```
{{- if .Values.securityConfig.config.data -}}
apiVersion: v1
kind: Secret
metadata:
  name: {{ include "opensearch.uname" . }}-securityconfig
```

So if chart is using exist secret, securityConfig.config.data can't be used. As result "securityconfig" secret will not be generated and securityconfigchecksum annotation will not be calculated.